src/async_fuse/memfs/cache/backend.rs

```diff
@@ -0,0 +1,493 @@
+
+        fs::remove_dir_all(backend_root).await.unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_write_append() {
+        let backend_root = format!("{BACKEND_ROOT}/write_append");
+        fs::create_dir_all(&backend_root).await.unwrap();
+        let backend = prepare_backend(&backend_root);
+
+        let backend_root = Path::new(&backend_root);
+
+        let mut block = Block::from_slice(BLOCK_SIZE_IN_BYTES, BLOCK_CONTENT);
+        block.set_start(0);
+        block.set_end(4);
+        backend.store(0, 0, block).await;
+        let loaded = backend.load(0, 0).await.unwrap();
+        assert_eq!(loaded.as_slice(), b"foo \0\0\0\0");
+
+        let mut block = Block::from_slice(BLOCK_SIZE_IN_BYTES, BLOCK_CONTENT);
+        block.set_start(4);
+        block.set_end(8);
+        backend.store(0, 0, block).await;
+        let loaded = backend.load(0, 0).await.unwrap();
+        assert_eq!(loaded.as_slice(), BLOCK_CONTENT);
+
+        fs::remove_dir_all(backend_root).await.unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_overwrite() {
+        let backend_root = format!("{BACKEND_ROOT}/overwrite");
+        fs::create_dir_all(&backend_root).await.unwrap();
+        let backend = prepare_backend(&backend_root);
+
+        let backend_root = Path::new(&backend_root);
+
+        let block = Block::from_slice(BLOCK_SIZE_IN_BYTES, BLOCK_CONTENT);
+        backend.store(0, 0, block).await;
+        let loaded = backend.load(0, 0).await.unwrap();
+        assert_eq!(loaded.as_slice(), BLOCK_CONTENT);
+
+        let block = Block::from_slice_with_range(BLOCK_SIZE_IN_BYTES, 4, 8, b"foo ");
+        backend.store(0, 0, block).await;
+        let loaded = backend.load(0, 0).await.unwrap();
+        assert_eq!(loaded.as_slice(), b"foo foo ");
+
+        fs::remove_dir_all(backend_root).await.unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_truncate() {
+        let backend_root = format!("{BACKEND_ROOT}/truncate");
+        fs::create_dir_all(&backend_root).await.unwrap();
+        let backend = prepare_backend(&backend_root);
+
+        let backend_root = Path::new(&backend_root);
+
+        for block_id in 0..4 {
+            backend
+                .store(
+                    0,
+                    block_id,
+                    Block::from_slice(BLOCK_SIZE_IN_BYTES, BLOCK_CONTENT),
+                )
+                .await;
+        }
+
+        backend.truncate(0, 4, 1, BLOCK_SIZE_IN_BYTES).await;
+        let loaded = backend.load(0, 0).await.unwrap();
+        assert_eq!(loaded.as_slice(), BLOCK_CONTENT);
+
+        for block_id in 1..4 {
+            let loaded = backend.load(0, block_id).await;
+            assert!(loaded.is_none());
+        }
+
+        backend.truncate(0, 1, 0, BLOCK_SIZE_IN_BYTES).await;
+
+        let file_path = backend_root.join("0");
+        let file_dir_exist = fs::try_exists(&file_path).await.unwrap();
+        assert!(!file_dir_exist);
```

Comment:
What's purpose of this block?

Reply:
If a file is truncated to `block_id = 0`, the whole file will be removed from the storage. If so, the directory representing this file will be removed directly. And this block is to test if the whole dir is removed properly.

I'm sorry for the unclear codes, I will write more comments for the test cases.